adapters_test.go

```diff
@@ -0,0 +1,355 @@
+
+			adaptedTxs, err := monomer.AdaptCosmosTxsToEthTxs(txs)
+			require.NoError(t, err)
+			require.Equal(t, len(ethTxs), len(adaptedTxs))
+			for i := range ethTxs {
+				require.Equal(t, ethTxs[i].Hash(), adaptedTxs[i].Hash(), i)
+			}
+		})
+	}
+
+	t.Run("unmarshal cosmos tx error", func(t *testing.T) {
+		_, err := monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{[]byte("invalid")})
+		require.Error(t, err)
+	})
+
+	t.Run("unexpected number of msgs in Eth Cosmos tx error", func(t *testing.T) {
+		sdkTx := sdktx.Tx{
+			Body: &sdktx.TxBody{
+				Messages: []*codectypes.Any{},
+			},
+		}
+
+		depositSDKMsgBytes, err := sdkTx.Marshal()
+		require.NoError(t, err)
+
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("L1 Attributes tx not found error", func(t *testing.T) {
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{})
+		require.NoError(t, err)
+
+		sdkTx := sdktx.Tx{
+			Body: &sdktx.TxBody{
+				Messages: []*codectypes.Any{msgAny},
+			},
+		}
+
+		depositSDKMsgBytes, err := sdkTx.Marshal()
+		require.NoError(t, err)
+
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("unmarshal MsgL1Txs smsg error", func(t *testing.T) {
+		sdkTx := sdktx.Tx{
+			Body: &sdktx.TxBody{
+				Messages: []*codectypes.Any{
+					{
+						Value: []byte("invalid"),
+					},
+				},
+			},
+		}
+
+		depositSDKMsgBytes, err := sdkTx.Marshal()
+		require.NoError(t, err)
+
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("MsgL1Tx contains non-deposit tx error", func(t *testing.T) {
+		nonDepNum := 5
+		inclusionListTxs, _, _ := generateTxsAndTxsBytes(t, 0, nonDepNum)
+
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{
+			TxBytes: inclusionListTxs,
+		})
+		require.NoError(t, err)
+
+		sdkTx := sdktx.Tx{
+			Body: &sdktx.TxBody{
+				Messages: []*codectypes.Any{msgAny},
+			},
+		}
+
+		depositSDKMsgBytes, err := sdkTx.Marshal()
+		require.NoError(t, err)
+
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+
+	t.Run("unmarshal binary error", func(t *testing.T) {
+		msgAny, err := codectypes.NewAnyWithValue(&rolluptypes.MsgApplyL1Txs{
+			TxBytes: [][]byte{[]byte("invalid")},
+		})
+		require.NoError(t, err)
+
+		sdkTx := sdktx.Tx{
+			Body: &sdktx.TxBody{
+				Messages: []*codectypes.Any{msgAny},
+			},
+		}
+
+		depositSDKMsgBytes, err := sdkTx.Marshal()
+		require.NoError(t, err)
+
+		_, err = monomer.AdaptCosmosTxsToEthTxs(bfttypes.Txs{depositSDKMsgBytes})
+		require.Error(t, err)
+	})
+}
+
+func generateTxsAndTxsBytes(
+	t *testing.T,
+	depTxNum int,
+	nonDepTxNum int,
+) ([][]byte, ethtypes.Transactions, []hexutil.Bytes) {
+	_, depositTx, cosmosEthTx := testutils.GenerateEthTxs(t)
+	inclusionListTxsBytes := make([][]byte, 0, depTxNum+nonDepTxNum)
```

Comment:
is `txBytes` a better name here? Not sure we need the `inclusionList` prefix

Reply:
[Rename variables](https://github.com/polymerdao/monomer/pull/157/commits/c748d5a6a7c8e280630e1ea8740ad76f5aa5c7f2)